modules/tinymce/src/themes/silver/main/ts/ui/toolbar/button/ToolbarButtons.ts

```diff
@@ -87,30 +103,50 @@ const renderCommonStructure = (
   behaviours: Optional<Behaviours>,
   providersBackstage: UiFactoryBackstageProviders
 ): AlloyButtonSpec => {
+  const optMemDisplayText = text.map(
+    (text) => Memento.record(renderLabel(text, ToolbarButtonClasses.Button, providersBackstage))
+  );
+  const optMemDisplayIcon = icon.map(
+    (icon) => Memento.record(renderReplaceableIconFromPack(icon, providersBackstage.icons))
+  );
   return {
     dom: {
       tag: 'button',
       classes: [ ToolbarButtonClasses.Button ].concat(text.isSome() ? [ ToolbarButtonClasses.MatchWidth ] : []),
       attributes: getTooltipAttributes(tooltip, providersBackstage)
     },
     components: componentRenderPipeline([
-      icon.map((iconName) => renderIconFromPack(iconName, providersBackstage.icons)),
-      text.map((text) => renderLabel(text, ToolbarButtonClasses.Button, providersBackstage))
+      optMemDisplayIcon.map((mem) => mem.asSpec()),
+      optMemDisplayText.map((mem) => mem.asSpec()),
     ]),
 
     eventOrder: {
       [NativeEvents.mousedown()]: [
         'focusing',
         'alloy.base.behaviour',
         'common-button-display-events'
-      ]
+      ],
+      [SystemEvents.attachedToDom()]: [ 'common-button-display-events', 'reflecting' ]
     },
 
     buttonBehaviours: Behaviour.derive(
       [
         DisablingConfigs.toolbarButton(providersBackstage.isDisabled),
         ReadOnly.receivingConfig(),
         AddEventsBehaviour.config('common-button-display-events', [
+          AlloyEvents.runOnAttached((comp, _se) => {
+            Css.set(comp.element, 'width', Css.get(comp.element, 'width') );
```

Comment:
Maybe add a comment why this is required since we want to keep the width when `setText` is executed but that might not be obvious if you read this out of context. Or maybe describe it with code since this is used in a few places maybe it's better to have say a function that does `ToolbarUtils.forceInitialSize(comp)` or similar that does the code and also describes why it's there.

```suggestion
            Css.set(comp.element, 'width', Css.get(comp.element, 'width'));
```

Reply:
Done. I used UiUtils since it is not used only in toolbar but also in dropdown.